### PR TITLE
Simplify performance measurements

### PR DESCRIPTION
**Pull request by oehme**
The many measurements that we injected into the build under test were
skewing our measurements to the point of making them unreliable or
unrealistic. We now only measure end to end build time. Here's a breakdown
with the rationale for removing each other measurement:

- configuration time: can be done by having a `gradle help` scenario instead
- execution time: the user does not care whether a long build is stuck in execution or configuration
- setup/teardown: was ill-defined anyway, basically total - configuration - execution
- JIT compile time: this is nothing we can influence and thus pointless to measure
- Memory usage: Was only measured at one point in the build, which doesn't tell us anything about
any problems at any other point in the build
- GC CPU time: If this increases we'd see it in total execution time

Generally, looking at the graphs has never pointed us directly at the problem, we always need to
profile anyway. So instead of skewing our measurements with lots of profling code, we should
instead use a dedicated profiling job to measure if we actually see a regression.

Memory usage can be tested indirectly by giving each scenario a reasonable amount of memory.
If memory usage rises above that reasonable limit, we'd see execution time rise, telling us about
the regression. Generally, we do not optimize for smallest memory usage, but for fastest execution
with reasonable memory overhead.

This change also removes all JVM tweaking and wait periods which we introduced in an attempt to
make tests more predictable and stable. These tweaks have not really helped us achieve more stable
tests and have often done the opposite. They also add lots of complexity and make our tests more
unrealistic. A real users will not add all these JVM options to Gradle.

### CODE REVIEW

**Comment by oehme** [comment]
There are still small pieces of cleanup to do (e.g. remove the now unused fields from the database), but I wanted to get this reviewed and merged soon, because it already reduces unstable tests to a minimum (0-2 failing tests per run)

The remaining unstability can be fixed with sane memory settings, which I'm working on next.

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/GradleVsMavenBuildPerformanceResults.groovy [hunk 1]
```diff
@@ -27,14 +27,13 @@ class GradleVsMavenBuildPerformanceResults extends CrossBuildPerformanceResults
...
-                println it
-            }
+            def stats = baselineVersion.getSpeedStatsAgainst("Gradle", gradle)
+
+            println stats
 
             def mavenIsFaster = baselineVersion.fasterThan(gradle)
             if (mavenIsFaster) {
-                throw new AssertionError(Object.cast(stats[0]))
+                throw new AssertionError(stats as Object)
```
Comment (melix) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/GradleVsMavenBuildPerformanceResults.groovy]: Uh? Is this to make the Groovy compiler happy? This looks very strange.

Comment (oehme) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/GradleVsMavenBuildPerformanceResults.groovy]: Yes, there is a constructor that takes a String, but it's private.

**Comment by oehme** [comment]
> We can replace with smarter limits for each scenario, indicating what we think is reasonable for such a scenario, though.

Yes, that's exactly what I'm working on next 👍 

> gradle help is only a good approximation of configuration time when configuration time is significant. For small builds this is not true, and help typically configures the software model and involves a lot more than it seems

You are right, small builds basically only measure "startup time". That's why we'll probably only keep a few of those in the long run and focus more on bigger projects with significant config time.

**Pull request merged**